[docs] eas update: better navigation titles

quinlanj opened the pull request:
# Why

Improve the readability of the navigation sidebar for EAS Update, similar to EAS Build

# How

- Most articles under the `EAS Update` section should not have `EAS Update` in the article title because it is already placed in the relevant category. EAS Build does this, and it makes it easier to navigate the docs.
- Avoid long titles. For example `How to optimize assets with EAS Update` can be shortened to `Optimizing Assets`

# UX
Before: 
![Screenshot 2023-06-19 at 5 02 14 PM](https://github.com/expo/expo/assets/6380927/6ebd9463-55cb-4721-a748-710a915df326)

After:
![Screenshot 2023-06-19 at 5 02 37 PM](https://github.com/expo/expo/assets/6380927/cf0aad51-bd08-4e2d-9c39-efdea3dc4a8d)


# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

quinlanj committed:
[docs] eas update: better page titles

kbrandwijk commented on docs/pages/eas-update/introduction.mdx:
```diff
@@ -30,7 +30,7 @@ Still using Classic Updates? We moved those docs to [the archive](/archive/class
 
 <BoxLink
-  title="Developing with EAS Update"
+  title="Developing Faster"
```
```suggestion
  title="Developing faster"
```

kbrandwijk commented on docs/pages/eas-update/eas-update-with-local-build.mdx:
```diff
@@ -1,5 +1,5 @@
 ---
-title: EAS Update with a local build
+title: Building Locally
```
```suggestion
title: Building locally
```

kbrandwijk commented on docs/pages/eas-update/developing-with-eas-update.mdx:
```diff
@@ -1,5 +1,5 @@
 ---
-title: Developing with EAS Update
+title: Developing Faster
```
```suggestion
title: Developing faster
```

kbrandwijk reviewed (commented):
Left a few suggestions for consistency with other titles throughout this and other sections.

Simek commented:
AFAIK we added the "EAS Update" to the page titles on purpose, to avoid duplication of the search results and to keep them clear and separated, since, for example, in the past "Debugging" query was returning few pages titled like that, and it was hard for the users to discern which part of the docs/which product etc. given page is devoted to.

kbrandwijk commented:
> AFAIK we added the "EAS Update" to the page titles on purpose, to avoid duplication of the search results and to keep them clear and separated, since, for example, in the past "Debugging" query was returning few pages titled like that, and it was hard for the users to discern which part of the docs/which product etc. given page is devoted to.

Then these are probably best implemented as `sidebar_title` overrides, leaving the page title as-is for search purposes. wdyt?

amandeepmittal commented:
We can also change the usage of gerunds to use short-form verbs. For example, "Optimizing assets" to "Optimize assets". This a pattern we've started following in the "Home" section of our docs: https://docs.expo.dev/

quinlanj committed:
pr feedback:

quinlanj commented:
Thanks for the feedback, everyone! I've updated the PR feedback with the following:
- used `sidebar_title`
- preserve `EAS Update` in the page titles, except for CodePush and Classic Updates migration docs, which shouldn't have duplicates since they are only relevant to EAS Update
- modified the page titles and sidebar titles to use short-form verbs (ie) "Optimizing assets" to "Optimize assets"

kbrandwijk reviewed (approved):
Thanks for the fixes! :shipit:
